src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -212,7 +212,7 @@
   <Target Name="CheckTestPlatforms"
           AfterTargets="Test">
     <GetTargetMachineInfo>
-      <Output TaskParameter="TargetOS" PropertyName="TargetOS" />
+      <Output TaskParameter="TargetOS" PropertyName="TargetOS" Condition="'$(TargetOS)' == ''" />
```

Comment:
I guess I was expecting this to be set here https://github.com/dotnet/buildtools/blob/b1e5912fb92c347cede8c9471eff23011e6c8419/src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets#L82

But I see that is guarded by TestWithLocalLibraries I'm thinking we should just removed that condition and always use that to setup TargetOS.